src/poseidon_gencontract.js

```diff
@@ -9,92 +9,91 @@ import { ethers } from "ethers";
 
 import poseidonConstants from "./poseidon_constants.js";
 
-const { C:K, M } = unstringifyBigInts(poseidonConstants);
+
+// ONLY for Poseidon3 per starkwares implementation/configs https://github.com/starkware-industries/poseidon
+const { C: K, M } = unstringifyBigInts(poseidonConstants);
 
 const N_ROUNDS_F = 8;
-const N_ROUNDS_P = [56, 57, 56, 60, 60, 63, 64, 63];
+const N_ROUNDS_P = 83;
 
 function toHex256(a) {
     let S = a.toString(16);
-    while (S.length < 64) S="0"+S;
+    while (S.length < 64) S = "0" + S;
     return "0x" + S;
 }
 
 export function createCode(nInputs) {
 
-    if (( nInputs<1) || (nInputs>8)) throw new Error("Invalid number of inputs. Must be 1<=nInputs<=8");
-    const t = nInputs + 1;
+    if ((nInputs != 3)) throw new Error("Invalid number of inputs. Must be 3");
+    const t = nInputs;
     const nRoundsF = N_ROUNDS_F;
-    const nRoundsP = N_ROUNDS_P[t - 2];
+    const nRoundsP = N_ROUNDS_P;
 
     const C = new Contract();
 
     function saveM() {
-        for (let i=0; i<t; i++) {
-            for (let j=0; j<t; j++) {
-                C.push(toHex256(M[t-2][i][j]));
-                C.push((1+i*t+j)*32);
+        for (let i = 0; i < t; i++) {
+            for (let j = 0; j < t; j++) {
+                C.push(toHex256(M[t - 2][i][j]));
+                C.push((1 + i * t + j) * 32);
                 C.mstore();
             }
         }
     }
 
     function ark(r) {   // st, q
-        for (let i=0; i<t; i++) {
+        for (let i = 0; i < t; i++) {
             C.dup(t); // q, st, q
-            C.push(toHex256(K[t-2][r*t+i]));  // K, q, st, q
-            C.dup(2+i); // st[i], K, q, st, q
+            C.push(toHex256(K[t - 2][r * t + i]));  // K, q, st, q
+            C.dup(2 + i); // st[i], K, q, st, q
             C.addmod(); // newSt[i], st, q
             C.swap(1 + i); // xx, st, q
             C.pop();
         }
     }
 
     function sigma(p) {
-        // sq, q
+        // st, q
         C.dup(t);   // q, st, q
-        C.dup(1+p); // st[p] , q , st, q
+        C.dup(1 + p); // st[p] , q , st, q
         C.dup(1);   // q, st[p] , q , st, q
-        C.dup(0);   // q, q, st[p] , q , st, q
-        C.dup(2);   // st[p] , q, q, st[p] , q , st, q
-        C.dup(0);   // st[p] , st[p] , q, q, st[p] , q , st, q
-        C.mulmod(); // st2[p], q, st[p] , q , st, q
-        C.dup(0);   // st2[p], st2[p], q, st[p] , q , st, q
-        C.mulmod(); // st4[p], st[p] , q , st, q
-        C.mulmod(); // st5[p], st, q
-        C.swap(1+p);
+        C.dup(1);   // st[p] , q, st[p] , q , st, q
```

Comment:
This change here basically is changing it from pow (x, 5) to pow(x,3) Polygons used a higher degree s-box function.